common/lib/dependabot/dependency_graphers/base.rb

```diff
@@ -5,6 +5,21 @@
 
 module Dependabot
   module DependencyGraphers
+    # This is a small value class that specifies the information we expect to be returned for each
+    # dependency strictly.
+    class ResolvedDependency < T::Struct
```

Comment:
Is it possible to make this immutable?

```suggestion
    class ResolvedDependency < T::ImmutableStruct
```

Reply:
Good call, I think it should be, I'll check!